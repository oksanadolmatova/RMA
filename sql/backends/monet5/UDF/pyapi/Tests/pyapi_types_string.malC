
# basic strings
bstr:= bat.new(:str);
bat.append(bstr,"asdf":str);
bat.append(bstr,"sd asd asd asd asd a":str);
bat.append(bstr,"test123":str);
bat.append(bstr,"test":str);
rstr:bat[:str] := pyapi.eval(nil:ptr,"return(arg1)",bstr);
io.print(rstr);

# strings with nil value
bstr:= bat.new(:str);
bat.append(bstr,"asdf":str);
bat.append(bstr,"sd asd asd asd asd a":str);
bat.append(bstr,nil:str);
bat.append(bstr,"test":str);
rstr1:bat[:str] := pyapi.eval(nil:ptr,"return(arg1)",bstr);
io.print(rstr1);

# unicode strings
astr:= bat.new(:str);
bat.append(astr,"Hannes Mühleisen":str);
bat.append(astr,"héllo":str);
bat.append(astr,"éáú üüäãö":str);
bat.append(astr,"幺巾乡阜阝测试一些中国符号":str);
bat.append(astr,"いくつかの日本のシンボルをテストします":str);
rstr2:bat[:str] := pyapi.eval(nil:ptr,"return(arg1)",astr);
io.print(rstr2);

# return python strings
rstr3:bat[:str] := pyapi.eval(nil:ptr,"return([[\"Hello\", \"World\"]])");
io.print(rstr3);

# convert boolean to string
rstr4:bat[:str] := pyapi.eval(nil:ptr,"return([[True, True, False]])");
io.print(rstr4);

# convert bit to string
bbit:= bat.new(:bit);
bat.append(bbit,1:bit);
bat.append(bbit,0:bit);
rbit:bat[:str] := pyapi.eval(nil:ptr,"return(arg1)",bbit);
io.print(rbit);

# convert bte to string
bbte:= bat.new(:bte);
bat.append(bbte,42:bte);
bat.append(bbte,84:bte);
rbte:bat[:str] := pyapi.eval(nil:ptr,"return(arg1)",bbte);
io.print(rbte);

# convert short to string
bsht:= bat.new(:sht);
bat.append(bsht,42:sht);
bat.append(bsht,82:sht);
rsht:bat[:str] := pyapi.eval(nil:ptr,"return(arg1)",bsht);
io.print(rsht);

# convert int to string
bint:= bat.new(:int);
bat.append(bint,1804289383:int);
bat.append(bint,846930886:int);
bat.append(bint,1681692777:int);
bat.append(bint,1714636915:int);
bat.append(bint,nil:int);
rstr5:bat[:str] := pyapi.eval(nil:ptr,"return(arg1)",bint);
io.print(rstr5);

# convert long to string
blng:= bat.new(:lng);
bat.append(blng,1804289383L);
bat.append(blng,846930886L);
rlng:bat[:str] := pyapi.eval(nil:ptr,"return(arg1)",blng);
io.print(rlng);

# convert float to string
bflt:= bat.new(:flt);
bat.append(bflt,18042.89383:flt);
bat.append(bflt,846.930886:flt);
bat.append(bflt,16.81692777:flt);
bat.append(bflt,1714636.915:flt);
bat.append(bflt,nil:flt);
rstr6:bat[:str] := pyapi.eval(nil:ptr,"return(arg1)",bflt);
io.print(rstr6);

# convert dbl to string
bdbl:= bat.new(:dbl);
bat.append(bdbl,18042.89383:dbl);
bat.append(bdbl,846.930886:dbl);
bat.append(bdbl,16.81692777:dbl);
bat.append(bdbl,1714636.915:dbl);
bat.append(bdbl,nil:dbl);
rstr7:bat[:str] := pyapi.eval(nil:ptr,"return(arg1)",bdbl);
io.print(rstr7);

# convert string to bit
bstr:= bat.new(:str);
bat.append(bstr,"1":str);
bat.append(bstr,"0":str);
rbit8:bat[:bit] := pyapi.eval(nil:ptr,"return(arg1)",bstr);
io.print(rbit8);

# convert string to bte
bstr:= bat.new(:str);
bat.append(bstr,"42":str);
bat.append(bstr,"84":str);
rbte9:bat[:bte] := pyapi.eval(nil:ptr,"return(arg1)",bstr);
io.print(rbte9);

# convert string to short
bstr:= bat.new(:str);
bat.append(bstr,"42":str);
bat.append(bstr,"82":str);
rsht10:bat[:sht] := pyapi.eval(nil:ptr,"return(arg1)",bstr);
io.print(rsht10);

# convert string to int
bstr:= bat.new(:str);
bat.append(bstr,"1804289383":str);
bat.append(bstr,"846930886":str);
bat.append(bstr,"1681692777":str);
bat.append(bstr,"1714636915":str);
bat.append(bstr,nil:str);
rstr11:bat[:int] := pyapi.eval(nil:ptr,"return(arg1)",bstr);
io.print(rstr11);

# convert string to oid
bstr:= bat.new(:str);
bat.append(bstr,"18042893":str);
bat.append(bstr,"8469308":str);
rwrd13:bat[:oid] := pyapi.eval(nil:ptr,"return(arg1)",bstr);
io.print(rwrd13);

# convert string to long
bstr:= bat.new(:str);
bat.append(bstr,"1804289383":str);
bat.append(bstr,"846930886":str);
rlng14:bat[:lng] := pyapi.eval(nil:ptr,"return(arg1)",bstr);
io.print(rlng14);

# convert string to float
bstr:= bat.new(:str);
bat.append(bstr,"18042.89383":str);
bat.append(bstr,"846.930886":str);
bat.append(bstr,"16.81692777":str);
bat.append(bstr,"1714636.915":str);
bat.append(bstr,nil:str);
rstr15:bat[:flt] := pyapi.eval(nil:ptr,"return(arg1)",bstr);
io.print(rstr15);

# convert string to dbl
bstr:= bat.new(:str);
bat.append(bstr,"18042.89383":str);
bat.append(bstr,"846.930886":str);
bat.append(bstr,"16.81692777":str);
bat.append(bstr,"1714636.915":str);
bat.append(bstr,nil:str);
rstr16:bat[:dbl] := pyapi.eval(nil:ptr,"return(arg1)",bstr);
io.print(rstr16);
