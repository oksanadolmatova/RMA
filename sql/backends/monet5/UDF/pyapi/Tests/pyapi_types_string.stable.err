stderr of test 'pyapi_types_string` in directory 'sql/backends/monet5/UDF/pyapi` itself:


# 20:45:38 >  
# 20:45:38 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=37361" "--set" "mapi_usock=/var/tmp/mtest-9070/.s.monetdb.37361" "--set" "monet_prompt=" "--forcemito" "--set" "mal_listing=2" "--dbpath=/home/mytherin/opt/var/MonetDB/mTests_monetdb5_extras_pyapi" "--set" "mal_listing=2" "--set" "embedded_r=true" "--set" "embedded_py=true"
# 20:45:38 >  

# builtin opt 	gdk_dbpath = /home/mytherin/opt/var/monetdb5/dbfarm/demo
# builtin opt 	gdk_debug = 0
# builtin opt 	gdk_vmtrim = no
# builtin opt 	monet_prompt = >
# builtin opt 	monet_daemon = no
# builtin opt 	mapi_port = 50000
# builtin opt 	mapi_open = false
# builtin opt 	mapi_autosense = false
# builtin opt 	sql_optimizer = default_pipe
# builtin opt 	sql_debug = 0
# cmdline opt 	gdk_nr_threads = 0
# cmdline opt 	mapi_open = true
# cmdline opt 	mapi_port = 37361
# cmdline opt 	mapi_usock = /var/tmp/mtest-9070/.s.monetdb.37361
# cmdline opt 	monet_prompt = 
# cmdline opt 	mal_listing = 2
# cmdline opt 	gdk_dbpath = /home/mytherin/opt/var/MonetDB/mTests_monetdb5_extras_pyapi
# cmdline opt 	mal_listing = 2
# cmdline opt 	embedded_r = true
# cmdline opt 	embedded_py = true
# cmdline opt 	gdk_debug = 536870922

# 20:45:39 >  
# 20:45:39 >  "mclient" "-lmal" "-ftest" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-9070" "--port=37361"
# 20:45:39 >  


# 20:45:39 >  
# 20:45:39 >  "Done."
# 20:45:39 >  

