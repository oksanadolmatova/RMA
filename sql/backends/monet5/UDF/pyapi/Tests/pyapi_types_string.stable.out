stdout of test 'pyapi_types_string` in directory 'sql/backends/monet5/UDF/pyapi` itself:


# 20:45:38 >  
# 20:45:38 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=37361" "--set" "mapi_usock=/var/tmp/mtest-9070/.s.monetdb.37361" "--set" "monet_prompt=" "--forcemito" "--set" "mal_listing=2" "--dbpath=/home/mytherin/opt/var/MonetDB/mTests_monetdb5_extras_pyapi" "--set" "mal_listing=2" "--set" "embedded_r=true" "--set" "embedded_py=true"
# 20:45:38 >  

# MonetDB 5 server v11.22.0
# This is an unreleased version
# Serving database 'mTests_monetdb5_extras_pyapi', using 8 threads
# Compiled for x86_64-unknown-linux-gnu/64bit with 64bit OIDs and 128bit integers dynamically linked
# Found 7.684 GiB available main-memory.
# Copyright (c) 1993-July 2008 CWI.
# Copyright (c) August 2008-2015 MonetDB B.V., all rights reserved
# Visit http://www.monetdb.org/ for further information
# Listening for connection requests on mapi:monetdb://mytherin-N750JV:37361/
# Listening for UNIX domain connection requests on mapi:monetdb:///var/tmp/mtest-9070/.s.monetdb.37361
# Start processing logs sql/sql_logs version 52200
# Start reading the write-ahead log 'sql_logs/sql/log.4'
# Finished reading the write-ahead log 'sql_logs/sql/log.4'
# Finished processing logs sql/sql_logs
# MonetDB/SQL module loaded
# MonetDB/Python module loaded
# MonetDB/R   module loaded

Ready.

# 20:45:39 >  
# 20:45:39 >  "mclient" "-lmal" "-ftest" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-9070" "--port=37361"
# 20:45:39 >  

#io.print(rstr);
#--------------------------#
# h	t  # name
# void	str  # type
#--------------------------#
[ 0@0,	"asdf"	]
[ 1@0,	"sd asd asd asd asd a"	]
[ 2@0,	"test123"	]
[ 3@0,	"test"	]
#io.print(rstr);
#--------------------------#
# h	t  # name
# void	str  # type
#--------------------------#
[ 0@0,	"asdf"	]
[ 1@0,	"sd asd asd asd asd a"	]
[ 2@0,	nil	]
[ 3@0,	"test"	]
#io.print(rstr);
#--------------------------#
# h	t  # name
# void	str  # type
#--------------------------#
[ 0@0,	"Hannes Mühleisen"	]
[ 1@0,	"héllo"	]
[ 2@0,	"éáú üüäãö"	]
[ 3@0,	"幺巾乡阜阝测试一些中国符号"	]
[ 4@0,	"いくつかの日本のシンボルをテストします"	]
#io.print(rstr);
#--------------------------#
# h	t  # name
# void	str  # type
#--------------------------#
[ 0@0,	"Hello"	]
[ 1@0,	"World"	]
#io.print(rstr);
#--------------------------#
# h	t  # name
# void	str  # type
#--------------------------#
[ 0@0,	"1"	]
[ 1@0,	"1"	]
[ 2@0,	"0"	]
#io.print(rbit);
#--------------------------#
# h	t  # name
# void	str  # type
#--------------------------#
[ 0@0,	"1"	]
[ 1@0,	"0"	]
#io.print(rbte);
#--------------------------#
# h	t  # name
# void	str  # type
#--------------------------#
[ 0@0,	"42"	]
[ 1@0,	"84"	]
#io.print(rsht);
#--------------------------#
# h	t  # name
# void	str  # type
#--------------------------#
[ 0@0,	"42"	]
[ 1@0,	"82"	]
#io.print(rstr);
#--------------------------#
# h	t  # name
# void	str  # type
#--------------------------#
[ 0@0,	"1804289383"	]
[ 1@0,	"846930886"	]
[ 2@0,	"1681692777"	]
[ 3@0,	"1714636915"	]
[ 4@0,	nil	]
#io.print(rwrd);
#--------------------------#
# h	t  # name
# void	str  # type
#--------------------------#
[ 0@0,	"1804289383"	]
[ 1@0,	"846930886"	]
#--------------------------#
# h	t  # name
# void	str  # type
#--------------------------#
[ 0@0,	"18042.894531"	]
[ 1@0,	"846.930908"	]
[ 2@0,	"16.816927"	]
[ 3@0,	"1714636.875000"	]
[ 4@0,	nil	]
#io.print(rstr);
#--------------------------#
# h	t  # name
# void	str  # type
#--------------------------#
[ 0@0,	"18042.893830"	]
[ 1@0,	"846.930886"	]
[ 2@0,	"16.816928"	]
[ 3@0,	"1714636.915000"	]
[ 4@0,	nil	]
#io.print(rbit);
#--------------------------#
# h	t  # name
# void	bit  # type
#--------------------------#
[ 0@0,	true	]
[ 1@0,	false	]
#io.print(rbte);
#--------------------------#
# h	t  # name
# void	bte  # type
#--------------------------#
[ 0@0,	42	]
[ 1@0,	84	]
#io.print(rsht);
#--------------------------#
# h	t  # name
# void	sht  # type
#--------------------------#
[ 0@0,	42	]
[ 1@0,	82	]
#io.print(rstr);
#--------------------------#
# h	t  # name
# void	int  # type
#--------------------------#
[ 0@0,	1804289383	]
[ 1@0,	846930886	]
[ 2@0,	1681692777	]
[ 3@0,	1714636915	]
[ 4@0,	nil	]
#--------------------------#
# h	t  # name
# void	oid  # type
#--------------------------#
[ 0@0,	18042893@0	]
[ 1@0,	8469308@0	]
#io.print(rlng);
#--------------------------#
# h	t  # name
# void	lng  # type
#--------------------------#
[ 0@0,	1804289383	]
[ 1@0,	846930886	]
#io.print(rstr);
#--------------------------#
# h	t  # name
# void	flt  # type
#--------------------------#
[ 0@0,	18042.895	]
[ 1@0,	846.9309	]
[ 2@0,	16.816927	]
[ 3@0,	1714636.9	]
[ 4@0,	nil	]
#io.print(rstr);
#--------------------------#
# h	t  # name
# void	dbl  # type
#--------------------------#
[ 0@0,	18042.89383	]
[ 1@0,	846.930886	]
[ 2@0,	16.81692777	]
[ 3@0,	1714636.915	]
[ 4@0,	nil	]

# 20:45:39 >  
# 20:45:39 >  "Done."
# 20:45:39 >  

